src/utilities/mongo.ts

```diff
@@ -1,8 +1,8 @@
 import envVar from 'env-var';
 import { type Connection, createConnection } from 'mongoose';
 
-export async function createMongooseConnectionFromEnv(): Promise<Connection> {
+export function createMongooseConnectionFromEnv(): Connection {
   const mongoUri = envVar.get('MONGODB_URI').required().asString();
 
-  return createConnection(mongoUri).asPromise();
```

Comment:
This is the ultimate source of the pain I experienced with this task.

Basically, the original version was blocking the start of the server until the DB connection was established, but Knative expects the server to start listening ASAP, even if it's not ready (even if the health check will time out or return 50X responses).

Note that the both the old and new version of this function returns a Mongoose `Connection` instance. Any DB query will still hang until the connection is established or the timeout is hit. So, exactly the same behaviour as before.